### ISSUE TITLE
 Custom converters in mapping that returns null.

### ISSUE THREAD

**Issue by chrhol** [Starcounter/Starcounter#3623]
Separated from #3606

Adding a URI-mapping with a custom converter that returns `null`, leads to that the partial handler is never called.

In this sample, the `Self.GET` request (line 10) to the handler `/Sample1/partials/test/{?}` will never call the specific handler. This is due to the custom converter in the uri-mapping that returns `null`

I'm not sure what the correct behaviour should be here, or where the problem lies. Is it a bug in Starcounter code or is this expected behaviour (maybe even wanted behaviour?) and it is up to the one who adds the mapping to consider?

However the result will be the same as in #3622, the app isolation might be broken.

I think this needs to be discussed and (depending on the outcome of course) documented.

**Sample1 app** 
```c#
using Simplified.Ring1;
using Starcounter;

namespace Sample1 {
    class Program {
        static void Main() {
            Handle.GET("/Sample1/test", () => {
                dynamic json = new Json();
                json.Header = "Mapping test";
                json.Current = Self.GET<Json>("/Sample1/partials/test/FF");
                return json;
            });

            Handle.GET("/Sample1/partials/test/{?}", (string id) => {
                dynamic json = new Json();
                json.ID = id;
                return json;
            });

            UriMapping.OntologyMap("/Sample1/partials/test/{?}", typeof(Something).FullName, (id) => { return null; }, (id) => { return null; });
        }
    }
}
```

Output (in browser) from calling uri `/Sample1/test`

```javascript
{
    Header: "Mapping test",
    Current: { }
}
```

If the mapper (line 20 in the first sample) is changed so that no custom converters are used (`UriMapping.OntologyMap("/Sample1/partials/test/{?}", typeof(Something).FullName);`) instead the partial handler is called leading to the following output:

```javascript
{
    Header: "Mapping test",
    Current: {
        ID: "FF"
    }
}
```

/cc: @alemoi, @warpech,  @miyconst

**Comment by miyconst** [comment]
I think this is the correct behavior. Me and @warpech used it before. I don't remember exact class names, but the idea is that: Images app maps to `Person` class, and then the ID converter checks whether a Person has an Image, returns `NULL` if no.

Also I can think up lots more examples where this would be a useful feature, and I don't see any drawbacks. If someone would really need a handler to be called even if the ID does not exist, then just return `-1` and handle it.

The only thing should be fixed is the apps isolation.

**Comment by chrhol** [comment]
Thanks @miyconst for the answer!

Then this is the correct behaviour and the problem is that the wrong response is returned (same as #3622). I will start investigate the other issue, and see if solving that one also fixes this one.

**Comment by chrhol** [comment]
After the fix for #3622, the problem in this issue is also solved. Returning null from a custom converter will return null back as the response, instead of another apps response.
